openshift/pgslice/docker/partition_and_archive.sh

```diff
@@ -49,17 +49,26 @@ then
     exit 1
 fi
 
-export PGSLICE_URL = "postgresql://${PG_USER}:${PG_PASSWORD}@${PG_HOSTNAME}:${PG_PORT}/${PG_DATABASE}"
+ENCODED_PASS=$(python3 -c "import urllib.parse; print(urllib.parse.quote('${PG_PASSWORD}'))")
+PGSLICE_URL=postgresql://${PG_USER}:${ENCODED_PASS}@${PG_HOSTNAME}:${PG_PORT}/${PG_DATABASE}
 # Add partitions to the intermediate table (assumes it already exists)
-pgslice add_partitions $TABLE --intermediate --future 1
+pgslice add_partitions $TABLE --intermediate --future 1 --url $PGSLICE_URL
 # Fill the partitions with data from the original table
-pgslice fill $TABLE
+pgslice fill $TABLE --url $PGSLICE_URL
 # Analyze for query planner
-pgslice analyze $TABLE
+pgslice analyze $TABLE --url $PGSLICE_URL
 # Swap the intermediate table with the original table
-pgslice swap $TABLE
+pgslice swap $TABLE --url $PGSLICE_URL
 # Fill the rest (rows inserted between the first fill and the swap)
-pgslice fill $TABLE --swapped
+pgslice fill $TABLE --swapped --url $PGSLICE_URL
+
+
 # Dump any retired tables to S3 and drop
-pg_dump -c -Fc -t ${TABLE}_retired $PGSLICE_URL | gzip | AWS_ACCESS_KEY_ID="${AWS_ACCESS_KEY}" AWS_SECRET_ACCESS_KEY="${AWS_SECRET_KEY}" aws --endpoint="https://${AWS_HOSTNAME}" s3 cp - "s3://${AWS_BUCKET}/retired/${TABLE}_retired.dump.gz"
+# borrowing a lot from https://github.com/BCDevOps/backup-container
+_timestamp=`date +\%Y-\%m-\%d_%H-%M-%S`
+_datestamp=`date +\%Y/\%m`
+_target_filename="${PG_HOSTNAME}_${TABLE}_retired_${_timestamp}.sql.gz"
```

Comment:
`$TABLE` includes the month/year of the table being retired right?

Reply:
Correct!